core/src/main/kotlin/com/strumenta/kolasu/parsing/Parsing.kt

```diff
@@ -167,7 +173,10 @@ abstract class KolasuParser<R : Node, P : Parser, C : ParserRuleContext> {
 
     protected abstract fun createANTLRLexer(inputStream: InputStream): Lexer
     protected abstract fun createANTLRParser(tokenStream: TokenStream): P
-    protected abstract fun invokeRootRule(parser: P): C?
+    protected open fun invokeRootRule(parser: P): C? {
+        val entryPoint = parser::class.memberFunctions.find { it.name == parser.ruleNames[0] }
```

Comment:
nice idea

Reply:
We could add a comment on this method, to explain users when they should override this method

Reply:
OK, makes sense